pkg/planner/core/expression_rewriter.go

```diff
@@ -1367,17 +1369,19 @@ func (er *expressionRewriter) Leave(originInNode ast.Node) (retNode ast.Node, ok
 		}
 		er.ctxStackAppend(value, types.EmptyName)
 	case *driver.ParamMarkerExpr:
-		var value *expression.Constant
-		value, er.err = expression.ParamMarkerExpression(er.sctx, v, false)
-		if er.err != nil {
-			return retNode, false
-		}
-		initConstantRepertoire(value)
-		er.adjustUTF8MB4Collation(value.RetType)
-		if er.err != nil {
-			return retNode, false
-		}
-		er.ctxStackAppend(value, types.EmptyName)
+		withPlanCtx(func(planCtx *exprRewriterPlanCtx) {
```

Comment:
make `ParamMarkerExpr`  not allowed for `BuildExprWithAst` to make sure `er.sctx` we need fewer fields in `er.sctx`